x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/enrollment/CreateEnrollmentTokenTests.java

```diff
@@ -0,0 +1,308 @@
+            any(CheckedFunction.class))).thenReturn(httpResponseNotOK);
+
+        final CreateEnrollmentToken createEnrollmentToken = new CreateEnrollmentToken(environment, client);
+        IllegalStateException ex = expectThrows(IllegalStateException.class, () -> createEnrollmentToken.create("elastic",
+            new SecureString("elastic"), new SecureString("password")));
+        assertThat(ex.getMessage(), Matchers.containsString("Unexpected response code [400] from calling POST "));
+    }
+
+    public void testFailedRetrieveHttpInfo() throws Exception {
+        final CommandLineHttpClient client = mock(CommandLineHttpClient.class);
+        when(client.getDefaultURL()).thenReturn("http://localhost:9200");
+        final URL defaultUrl = new URL(client.getDefaultURL());
+        final URL createAPIKeyURL = CreateEnrollmentToken.createAPIKeyURL(defaultUrl);
+        final URL getHttpInfoURL = CreateEnrollmentToken.getHttpInfoURL(defaultUrl);
+
+        final HttpResponse httpResponseOK = new HttpResponse(HttpURLConnection.HTTP_OK, new HashMap<String, Object>());
+        when(client.execute(anyString(), any(URL.class), anyString(), any(SecureString.class), any(CheckedSupplier.class),
+            any(CheckedFunction.class))).thenReturn(httpResponseOK);
+
+        String createApiKeyResponseBody;
+        try (XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON)) {
+            builder.startObject()
+                .field("id", "DR6CzXkBDf8amV_48yYX")
+                .field("name", "enrollment_token_API_key_VuaCfGcBCdbkQm")
+                .field("expiration", "1622652381786")
+                .field("api_key", "x3YqU_rqQwm-ESrkExcnOg")
+                .endObject();
+            createApiKeyResponseBody = Strings.toString(builder);
+        }
+        when(client.execute(eq("POST"), eq(createAPIKeyURL), eq(ElasticUser.NAME), any(SecureString.class),
+            any(CheckedSupplier.class), any(CheckedFunction.class)))
+            .thenReturn(createHttpResponse(HttpURLConnection.HTTP_OK, createApiKeyResponseBody));
+
+        final HttpResponse httpResponseNotOK = new HttpResponse(HttpURLConnection.HTTP_BAD_REQUEST, new HashMap<String, Object>());
+        when(client.execute(anyString(), eq(getHttpInfoURL), anyString(), any(SecureString.class), any(CheckedSupplier.class),
+            any(CheckedFunction.class))).thenReturn(httpResponseNotOK);
+
+        final CreateEnrollmentToken createEnrollmentToken = new CreateEnrollmentToken(environment, client);
+        IllegalStateException ex = expectThrows(IllegalStateException.class, () -> createEnrollmentToken.create("elastic",
+            new SecureString("elastic"), new SecureString("password")));
+        assertThat(ex.getMessage(), Matchers.containsString("Unexpected response code [400] from calling GET "));
+    }
+
+    public void testNoKeyStore() throws Exception {
```

Comment:
maybe you can unit test getFingerprint instead ?

Reply:
getFingerprint is not static and depends on enviroment, so I am not sure we will win much here